test/UnitSaveTorture.cpp

```diff
@@ -106,6 +114,60 @@ namespace {
     }
 }
 
+void UnitSaveTorture::testModified()
+{
+    std::string name = "testModified";
+    std::string docName = "empty.ods";
+
+    std::string documentPath, documentURL;
+    helpers::getDocumentPathAndURL(docName, documentPath, documentURL, name);
+
+    TST_LOG("Starting test on " << documentURL << ' ' << documentPath);
+
+    std::shared_ptr<SocketPoll> poll = std::make_shared<SocketPoll>("WebSocketPoll");
+    poll->startThread();
+
+    Poco::URI uri(helpers::getTestServerURI());
+    auto wsSession = helpers::loadDocAndGetSession(poll, docName, uri, testname);
+
+    // It is vital that we can change the modified status successfully
+    // and also get correct notifications from the core for bgsave to work.
+    for (size_t i = 0; i < 4; ++i)
+    {
+        TST_LOG("modify document");
+        modifyDocument(wsSession);
+        LOK_ASSERT_EQUAL(waitForModifiedStatus(name, wsSession, std::chrono::seconds(3)), true);
+
+        std::string args = "{ \"Modified\": { \"type\": \"boolean\", \"value\": \"false\" } }";
+        TST_LOG("post force modified command: .uno:Modified " << args);
+        wsSession->sendMessage(std::string("uno .uno:Modified ") + args);
+
+        TST_LOG("wait for confirmation of (non-)modification:");
+        LOK_ASSERT_EQUAL(waitForModifiedStatus(name, wsSession, std::chrono::seconds(3)), false);
+    }
+
+    poll->joinThread();
+}
+
+namespace {
+    /*
+     * A sleep in a unit test !? but ... SfxBinding notifies its
+     * state changes around 250ms after they are made to reduce
+     * spamming clients; so - if we eg. do a save that forces an
```

Comment:
FWIW a thinking aloud throw away experiment on the SfxBinding timeouts: https://gerrit.libreoffice.org/c/core/+/167342